controllers/helmrepository_controller_oci.go

```diff
@@ -350,8 +355,30 @@ func (r *HelmRepositoryOCIReconciler) validateSource(ctx context.Context, obj *s
 	return sreconcile.ResultSuccess, nil
 }
 
-func loginOptionFromSecret(secret corev1.Secret) (registry.LoginOption, error) {
-	username, password := string(secret.Data["username"]), string(secret.Data["password"])
+// loginOptionFromSecret derives authentication data from a Secret to login to an OCI registry. This Secret
+// may either hold "username" and "password" fields or be of the corev1.SecretTypeDockerConfigJson type and hold
+// a corev1.DockerConfigJsonKey field with a complete Docker configuration. If both, "username" and "password" are
+// empty, a nil LoginOption and a nil error will be returned.
+func loginOptionFromSecret(registryURL string, secret corev1.Secret) (registry.LoginOption, error) {
```

Comment:
Any specific reason for this not to be an util function in the internal registry package?

Reply:
Discussed with Hidde and we'll create a new package `internal/helm/registry` and move the func there, together with the other OCI-specific utility code.

Reply:
Addressed in https://github.com/fluxcd/source-controller/pull/725/commits/116bd8c1b0cd8ec4b0432f65001bd91f294c53a0